### DISCUSSION TITLE
Unable to find a newly created employee in Sharepoint People search result.

### DISCUSSION THREAD

**Discussion by LAZMIC** [microsoft-search/pnp-modern-search#3411] [Q&A]
**Version used**
Ex: 4.9.3

**Describe the bug**
When using the main SharePoint search I can see the new employee in the query result. When Using the PnP Modern Search web part the same user does not show up in the Search Result. I can also find her in every other Microsoft App when search for said user. 

**To Reproduce**
Detailed steps to reproduce the behavior:
1. Go to your SharePoint Default Home
2. Use Search bar at the top of the page for the Newly created user.
3. Search Result will show the user
4. Search for the same new user via PnP Search box and no result will come up in the PnP Search Result

**Screen Shots**
![sp3](https://github.com/microsoft-search/pnp-modern-search/assets/92048530/9b73c633-40b1-4a0c-a959-5de596786391)
![sp2](https://github.com/microsoft-search/pnp-modern-search/assets/92048530/0d453ae7-653a-400c-a27a-a14391abc1de)
![sp](https://github.com/microsoft-search/pnp-modern-search/assets/92048530/4f8d6e19-7f73-466c-8466-478b53b3eb77)

**Expected behavior**
The search result should show new employees.

**Desktop (please complete the following information):**
 - Browser: Edge and Firefox

**Additional context**
Yes, the user has a valid Microsoft license. Yes, the user is a member of the Microsoft 365 group. 
I have check the result template to ensure there was no errors in code. The web part has been working as intended for Months. I'm also 90% sure this hasn't happen before. Our company has a high turnover rate and never experienced this issue before with other new employees.

**Comment by fernandaek** [comment]
We are experiencing the same issue: the newly added folder doesn’t show in the PnP modern search v4 and it is showing the deleted files.

**Comment by LAZMIC** [comment]
> We are experiencing the same issue: the newly added folder doesn’t show in the PnP modern search v4 and it is showing the deleted files.

Sorry to here that. Hope there is a solution out there.

**Comment by kasperbolarsen** [comment]
The search box in the suitbar/top is using Microsoft search as a data source,  and for people search PnP Modern Search is using SharePoint search which is based on User Profiles
1) check that the user exists in User Profiles 
2) use either SharePoint search query tool or SP Editor to verify if that profile is in the SharePoint Search index

**Comment by LAZMIC** [comment]
> The search box in the suitbar/top is using Microsoft search as a data source, and for people search PnP Modern Search is using SharePoint search which is based on User Profiles
> 
>     1. check that the user exists in User Profiles
> 
>     2. use either SharePoint search query tool or SP Editor to verify if that profile is in the SharePoint Search index

I'm guessing you are talking about the built-in people web part and yes the user can be found.
![sp4](https://github.com/microsoft-search/pnp-modern-search/assets/92048530/21b52888-b716-4527-b2a5-16a2ae41c74a)

**Comment by kasperbolarsen** [comment]
Hi

No, I am talking about using the PnP Modern Search searchbox and Results web part for people search as per the scenarios on the PnP Modern Search web site.

**Comment by LAZMIC** [comment]
Hmmm... I not sure what you mean. That is what my current setup is and the user is not showing up. I even tried to filter by the user's department and all employees of that department return but the new employee. I even used the search box to query the department name. This Result shows all user in the department except the new employee. I also spun up a mirror test site and add the PnP Modern Search web parts Search and results. Selected SharePoint Search > Selected localPeopleResults . Selected Handlebar "People" > connected the two parts. Same result.. No new employee in results.

Sorry for confusion.

**Comment by kasperbolarsen** [comment]
As you might know the datasource for SharePoint People search is the UPA ( User Profile Application ) in the SharePoint Admin center.
The data in the UPA can be synced from AD/AAD as per https://learn.microsoft.com/en-us/sharepoint/user-profile-sync

So I suggest that you check the UPA for that new user. If the new user is not in the UPA you have a sync issue, not a search issue.
If the new user is in the UPA I suggest you use one of the tools here , https://microsoft-search.github.io/pnp-modern-search/using-query-tools-to-verify-issues/ in order to check if the user is in the search index. If that is not the case, then you will have to wait as the search crawler is controlled by MS.
If the new user is in the search index then, and only then, it might be something to do with an issue in PnP Modern Search :-)
I have had several tenants today and yesterday where search was very slow to pick up changes so we might have bad performance right now.

**Comment by kasperbolarsen** [comment]
> We are experiencing the same issue: the newly added folder doesn’t show in the PnP modern search v4 and it is showing the deleted files.

This sounds like an unrelated issue as people search and folders are two very different objects. Please create a new issue and complete the template, thank you.

**Comment by LAZMIC** [comment]
> As you might know the datasource for SharePoint People search is the UPA ( User Profile Application ) in the SharePoint Admin center. The data in the UPA can be synced from AD/AAD as per https://learn.microsoft.com/en-us/sharepoint/user-profile-sync
> 
> So I suggest that you check the UPA for that new user. If the new user is not in the UPA you have a sync issue, not a search issue. If the new user is in the UPA I suggest you use one of the tools here , https://microsoft-search.github.io/pnp-modern-search/using-query-tools-to-verify-issues/ in order to check if the user is in the search index. If that is not the case, then you will have to wait as the search crawler is controlled by MS. If the new user is in the search index then, and only then, it might be something to do with an issue in PnP Modern Search :-) I have had several tenants today and yesterday where search was very slow to pick up changes so we might have bad performance right now.

Ok, now I'm understanding. Yes, when I go to SP Admin Center > More Features > User Profiles > Manage User Profiles. Search for the new user and the user profile can be found. i:0#.f\|membership\|username@domain

**Reply by kasperbolarsen** [reply]
So the next step is to check of the user exists in the search index

**Comment by wobba** [comment]
If data is not coming back, that is a data/API issue, not web part issue.
